tests/e2e/mgmt_wrkld_domain_isolation.go

```diff
@@ -554,4 +554,67 @@ var _ bool = ginkgo.Describe("[domain-isolation] Management-Workload-Domain-Isol
 			volumeSnapshot, pandoraSyncWaitTime, volHandle, snapshotId, true)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
+
+	/*
+	   Testcase-3
+	   Basic test
+	   Deploy statefulsets with 3 replica on namespace-1 in the supervisor cluster using shared policy with immediate volume binding mode storageclass.
+
+	   Steps:
+	   1. Create a wcp namespace and tag it to zone-3 workload zone.
+	   2. Read a shared storage policy which is tagged to wcp namespace created in step #1 using Immediate Binding mode.
+	   3. Create statefulset with replica count 3.
+	   4. Wait for PVC and PV to reach Bound state.
+	   5. Verify PVC has csi.vsphere.volume-accessible-topology annotation with all zones as its shared policy
+	   6. Verify PV has node affinity rule for all zones
+	   7. Verify statefulset pod is in up and running state.
+	   8. Veirfy Pod node annoation.
+	   9. Perform cleanup: Delete Statefulset
+	   10. Perform cleanup: Delete PVC
+	*/
+
+	ginkgo.It("Verifying volume creation and PV affinities with svc namespace tagged to shared policy, "+
+		"zone-3 tag, and immediate binding mode.", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// statefulset replica count
+		replicas = 3
+
+		// reading shared storage policy of zone-3 workload domain
+		storagePolicyName = GetAndExpectStringEnvVar(envSharedStoragePolicyName)
```

Comment:
Please check this, In your case you may need zonal specific policy. 'envSharedStoragePolicyName'  This may give common  tag based shared-policy